### PR TITLE
feat: swaps opencensus collector with otel based hypertrace-collector

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
- Fixes https://github.com/hypertrace/hypertrace/issues/179
- Our OpenTelemetry based [Hypertrace collector](https://github.com/hypertrace/hypertrace-collector) is ready to use now so replacing old opencensus based collector with it for both docker-compose and helm setup.

### Testing
I tested docker-compose locally and helm setup with EKS and seems working well. 

```
NAME                             TYPE           CLUSTER-IP       EXTERNAL-IP                                                              PORT(S)                                                                                         AGE
bootstrap                        ClusterIP      10.100.72.240    <none>                                                                   9092/TCP                                                                                        13h
hypertrace-collector             LoadBalancer   10.100.143.133   a9xxxxxxxxxxxxxxxxxxxxxxx1fb03a-17580528.us-west-2.elb.amazonaws.com    4317:32613/TCP,55681:32279/TCP,55678:30470/TCP,14268:32355/TCP,14250:32178/TCP,9411:30748/TCP   13h
hypertrace-data-config-service   ClusterIP      10.100.20.204    <none>                                                                   9012/TCP                                                                                        13h
hypertrace-data-query-service    ClusterIP      10.100.6.157     <none>                                                                   9001/TCP                                                                                        13h
hypertrace-graphql-service       ClusterIP      10.100.144.1     <none>                                                                   23431/TCP                                                                                       13h
hypertrace-ui                    LoadBalancer   10.100.85.3      adxxxxxxxxxxxxxxxxxxxx-311519380.us-west-2.elb.amazonaws.com   2020:31338/TCP

```

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
- https://github.com/hypertrace/hypertrace-collector
- https://github.com/open-telemetry/opentelemetry-collector

### CODE REVIEW

**Comment by pavolloffay** [comment]
As pointed in #179 shall we keep the old service `hypertrace-oc-collector ` for backwards compatibility or we don't care about it right now?

File: kubernetes/platform-services/values.yaml [hunk 1]
```diff
@@ -26,69 +26,86 @@ configServicePort: &configServicePort 9012
...
-    - name: grpc-opencensus
-      containerPort: 55678
-    - name: http-jaeger
-      containerPort: 14268
-    - name: jaeger-tchannel
-      containerPort: 14267
-    - name: zipkin
-      containerPort: 9411
+      cpu: 1
+      memory: 2Gi
```
Comment (rish691) [inline kubernetes/platform-services/values.yaml:33]: 2Gi, does this collector has higher mem requirements compared to previous?

**Comment by jcchavezs** [comment]
I would replace it as it would be weird to have a collector here and use another one.

File: kubernetes/platform-services/values.yaml (hunk 1 above)
Comment (JBAhire) [reply kubernetes/platform-services/values.yaml:33]: These are the default values we are using. I am not sure if @pavolloffay and @jcchavezs have tried tuning it even further. But yeah, we can eventually tune it as per our experience.

**Comment by JBAhire** [comment]
> I would replace it as it would be weird to have a collector here and use another one.

I agree. It makes sense to replace it as it is.

File: kubernetes/platform-services/values.yaml (hunk 1 above)
Comment (rish691) [reply kubernetes/platform-services/values.yaml:33]: No my point is around increasing that default value. For oc collector the memory limit was 256Mi, is there a reason behind increasing it to 2Gi?

Comment (JBAhire) [reply kubernetes/platform-services/values.yaml:33]: By default values I mean, default helm values for hypertrace collector which you can find here: https://github.com/hypertrace/hypertrace-collector/blob/main/helm/values.yaml

Comment (rish691) [reply kubernetes/platform-services/values.yaml:33]: Ah, I see

File: kubernetes/platform-services/values.yaml [hunk 2]
```diff
@@ -26,69 +26,86 @@ configServicePort: &configServicePort 9012
...
+        zipkin: 
+          endpoint: "0.0.0.0:9411"
+        jaeger:
+          protocols:
+            grpc: 
+              endpoint: "0.0.0.0:14250"
+            thrift_http: 
+              endpoint: "0.0.0.0:14268"
+        # Collect own metrics
+        prometheus:
```
Comment (kotharironak) [inline kubernetes/platform-services/values.yaml:86]: - No need to override this

File: kubernetes/platform-services/values.yaml [hunk 3]
```diff
@@ -26,69 +26,86 @@ configServicePort: &configServicePort 9012
...
+          config:
+            scrape_configs:
+              - job_name: "otel-collector"
+                scrape_interval: 10s
+                static_configs:
+                  - targets: ["0.0.0.0:8888"]
+      processors:
```
Comment (kotharironak) [inline kubernetes/platform-services/values.yaml:93]: no need to override this

File: kubernetes/platform-services/values.yaml [hunk 4]
```diff
@@ -26,69 +26,86 @@ configServicePort: &configServicePort 9012
...
-    kafka:
-      topics:
-        - name: jaeger-spans
-          replicationFactor: 3
-          partitions: 8
-          configs:
-            - retention.bytes=1073741824
-            - retention.ms=21600000
+          protocol_version: 2.0.0
+          brokers:
```
Comment (kotharironak) [inline kubernetes/platform-services/values.yaml:98]: no need to override this as well.

File: kubernetes/platform-services/values.yaml [hunk 5]
```diff
@@ -26,69 +26,86 @@ configServicePort: &configServicePort 9012
...
       receivers:
-        opencensus:
-          # keepalive settings can help load balancing, see receiver/README.md for more info.
-          keepalive:
-            server-parameters:
-              max-connection-age: 120s
-              max-connection-age-grace: 30s
-        jaeger: {}
-        zipkin: {}
+        otlp:
```
Comment (kotharironak) [inline kubernetes/platform-services/values.yaml:69]: The only thing we are the overriding receiver and exporter property
- https://github.com/hypertrace/hypertrace-collector/blob/main/helm/values.yaml
-

**Review by kotharironak — COMMENTED** [review]
LGTM, few minor comments

File: docker/postgres/docker-compose.yml [hunk 6]
```diff
@@ -27,12 +27,15 @@ services:
...
-    command: ["--config=/config/collector-config.yaml"]
+    command: ["--config", "/etc/opt/hypertrace/config.yml"]
+    volumes:
+      - ../configs/hypertrace-collector/config.yml:/etc/opt/hypertrace/config.yml:ro
     ports:
-      - 14268:14268 # Jaeger Thrift
-      - 14267:14267 # Jaeger HTTP
+      - 4317:4317 # grpc-otel
+      - 55681:55681 # http-otel
+      - 14268:14268 # Jaeger http
```
[the remark this answers is not in the document]
Comment (JBAhire) [reply docker/postgres/docker-compose.yml:38]: Jaeger Thrift/ Jaeger HTTP both are right conventions here. But as we are referring to 14268 as http in helm values(https://github.com/hypertrace/hypertrace-collector/blob/main/helm/values.yaml#L31) will stick to that convention here as well

File: docker/postgres/docker-compose.yml [hunk 7]
```diff
@@ -27,12 +27,15 @@ services:
 
   # Collects spans in different trace formats like Jaeger, Zipkin, etc
   hypertrace-collector:
-    image: hypertrace/hypertrace-oc-collector
+    image: hypertrace/hypertrace-collector:main
     container_name: hypertrace-collector
-    command: ["--config=/config/collector-config.yaml"]
+    command: ["--config", "/etc/opt/hypertrace/config.yml"]
```
[the remark this answers is not in the document]
Comment (JBAhire) [reply docker/postgres/docker-compose.yml:32]: Yes: https://github.com/hypertrace/hypertrace-collector/blob/main/Dockerfile

File: docker/docker-compose.yml [hunk 8]
```diff
@@ -30,12 +30,15 @@ services:
...
-    command: ["--config=/config/collector-config.yaml"]
+    command: ["--config", "/etc/opt/hypertrace/config.yml"]
+    volumes:
+      - ../docker/configs/hypertrace-collector/config.yml:/etc/opt/hypertrace/config.yml:ro
     ports:
-      - 14268:14268 # Jaeger Thrift
-      - 14267:14267 # Jaeger HTTP
+      - 4317:4317 # grpc-otel
+      - 55681:55681 # http-otel
+      - 14268:14268 # Jaeger http
```
[the remark this answers is not in the document]
Comment (JBAhire) [reply docker/docker-compose.yml:41]: https://github.com/hypertrace/hypertrace/pull/203#discussion_r584460432

**Review by kotharironak — APPROVED** [review]

**Pull request merged**